editor.planx.uk/src/@planx/components/TextInput/Public.tsx

```diff
@@ -31,6 +36,8 @@ const TextInputComponent: React.FC<Props> = (props) => {
     }),
   });
 
+  const characterCountLimit = characterCounterSwitch(props.type);
```

Comment:
I'd argue that this is logic which is should belong to the character count component.

 Could we run this check with the component, returning `null` if the condition is not met?